examples/main.cpp

```diff
@@ -0,0 +1,338 @@
+{
+    int ret = 0;
+    switch (handler->id)
+    {
+        case methods::invert:
+        {
+            CIL::invertColor(img);
+            break;
+        }
+        case methods::brightness:
+        {
+            function_parameters::percentage*
+                args = (function_parameters::percentage*) handler->data;
+
+            ret = IS_INVALID(args->percentage);
+            if (ret)
+                goto err;
+
+            CIL::changeBrightness(img, args->percentage);
+            break;
+        }
+        case methods::contrast:
+        {
+            function_parameters::percentage*
+                args = (function_parameters::percentage*) handler->data;
+
+            ret = IS_INVALID(args->percentage);
+            if (ret)
+                goto err;
+
+            CIL::changeContrast(img, args->percentage);
+            break;
+        }
+        case methods::crop:
+        {
+            function_parameters::dimensions*
+                args = (function_parameters::dimensions*) handler->data;
+
+            ret = IS_INVALID(args->length, args->breadth);
+            if (ret)
+                goto err;
+
+            CIL::cropImage(img,
+                           img.getDimensions({args->X_origin, args->Y_origin},
+                                             args->length, args->breadth));
+            break;
+        }
+        case methods::pad:
+        {
+            function_parameters::dimensions*
+                args = (function_parameters::dimensions*) handler->data;
+
+            ret = IS_INVALID(args->length, args->breadth);
+            if (ret)
+                goto err;
+
+            CIL::padImage(img,
+                          img.getDimensions({args->X_origin, args->Y_origin},
+                                            args->length, args->breadth));
+            break;
+        }
+        case methods::resize:
+        {
+            function_parameters::dimensions*
+                args = (function_parameters::dimensions*) handler->data;
+
+            ret = IS_INVALID(args->length, args->breadth);
+            if (ret)
+                goto err;
+
+            CIL::resize(img, args->length, args->breadth);
+            break;
+        }
+        case methods::rotate:
+        {
+            function_parameters::rotate* args = (function_parameters::rotate*)
+                                                    handler->data;
+
+            ret = IS_INVALID(args->degrees);
+            if (ret)
+                goto err;
+
+            CIL::rotate(img, args->degrees);
+            break;
+        }
+        case methods::flip:
+        {
+            function_parameters::flip* args = (function_parameters::flip*)
+                                                  handler->data;
+
+            ret = IS_INVALID(args->axis);
+            if (ret)
+                goto err;
+
+            CIL::flipImage(img, static_cast<CIL::Axis>(args->axis - 1));
```

Comment:
Should we add a check that `args->axis` has value 1 or 2?

Reply:
Fixed